source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -697,17 +697,33 @@ namespace AdaptiveNamespace
                 THROW_IF_FAILED(ellipseAsUIElement->put_Visibility(Visibility::Visibility_Collapsed));
                 // Handle ImageOpened event so we can check the imageSource's size to determine if it fits in its parent
                 EventRegistrationToken eventToken;
-                ComPtr<IInspectable> strongParentElement(parentElement);
+                ComPtr<IInspectable> localParentElement(parentElement);
+
+                // Take weak references to the ellipse and parent to avoid circular references between this lambda and
+                // its parents (Parent->Ellipse->ImageBrush->Lambda->(Parent and Ellipse))
+                WeakRef weakParent;
+                THROW_IF_FAILED(localParentElement.AsWeak(&weakParent));
+
+                WeakRef weakEllipse;
+                THROW_IF_FAILED(ellipseAsUIElement.AsWeak(&weakEllipse));
                 THROW_IF_FAILED(brushAsImageBrush->add_ImageOpened(
-                    Callback<IRoutedEventHandler>([ellipseAsUIElement, ellipseAsFrameworkElement, imageSourceAsBitmap, strongParentElement, isVisible](
+                    Callback<IRoutedEventHandler>([weakEllipse, imageSourceAsBitmap, weakParent, isVisible](
```

Comment:
>Callback [](start = 20, length = 8)

Hmm... perhaps we should audit all of our uses of `Callback` lambdas? e.g. `ArrangeButtonContent()` in ActionHelpers.cpp looks like it might have the same issue...

Reply:
(of course, you may have already done this)

---
In reply to: [331661884](https://github.com/microsoft/AdaptiveCards/pull/3507#discussion_r331661884) [](ancestors = 331661884)

Reply:
Good point. Logged #3511 to follow up.

---
In reply to: [331663216](https://github.com/microsoft/AdaptiveCards/pull/3507#discussion_r331663216) [](ancestors = 331663216,331661884)